src/traceradvectiondiffusion.jl

```diff
@@ -17,42 +19,113 @@ using GeophysicalFlows.MultiLayerQG: SingleLayerParams, TwoLayerParams, numberof
 import LinearAlgebra: mul!, ldiv!
 import GeophysicalFlows.MultiLayerQG
 
+# --
+# AdvectingFlows
+# --
+
+"Abstract super type for an advecting flow."
+abstract type AbstractAdvectingFlow end
+
+"""
+    struct  OneDAdvectingFlow <: AbstractAdvectingFlow
+
+A struct containing the advecting flow for a one dimensional `TracerAdvectionDiffusion.Problem`.
+Included is
+
+$(TYPEDFIELDS)
+"""
+struct OneDAdvectingFlow <: AbstractAdvectingFlow
+    "Function for the x-component of the advecting flow"
+             u :: Function
+    "Whether or not the flow is steady (i.e not time dependent)"
+    steadyflow :: Bool
+end
+
+"""
+    OneDAdvectingFlow(; u=noflow, steadyflow=false)
+
+Constructor for the `OneDAdvectingFlow`. The default function for the advecting flow component is `noflow`
+and `steadyflow=false`.    
+"""
+OneDAdvectingFlow(; u=noflow, steadyflow=false) = OneDAdvectingFlow(u, steadyflow)
```

Comment:
It's a bit counter-intuitive to have `noflow` (= zero flow) but at the same time `steadyflow=false`, isn't it?

Reply:
Yep it is! I will change this so that default is `noflow` and `steadyflow=false`.

Reply:
Do we even need `OneDAdvectingFlow.u` to be a function?

Reply:
Yes I did think we needed it to be. For the `ConstDiffTimeVaryingFlow` params it needs to be a function so that it can be computed at each time step and in the `ConstDiffSteadyFlow` params we compute the steady flow on the `grid` using a function so I thought it easiest to restrict `OneDAdvectingFlow.u`  to being a function.

Reply:
Yeap, I realised and took back my remark :)